### PR TITLE
Create sorbonne-student-law-review.csl

### PR DESCRIPTION
**Pull request by p-heckler**
Hi,
This is a style proposal for the Sorbonne Student Law Review. It is intended to be a multilingual version of the French legal citation structure since the review accepts submission in a range of languages. This style has been in the making for a while and I am hoping that it is up to the CSL standards!

### CODE REVIEW

**Comment by p-heckler** [comment]
Somehow, the generated citations here do not match the output I obtain when I test the file. Since I have not specified a `name-as-sort-order`, the given name should come first. I also have no idea where the parenthesis is coming from. Should I declare `name` explicitly or can I leave it as is, relying on the default settings? Also, the delimiter that follows quoted titles appears within the quote even though `quote="true"` is applied directly to the title. Again, I don't have this problem when testing it locally and I'm not sure where the code goes wrong here.

File: sorbonne-student-law-review.csl [hunk 1]
```diff
@@ -0,0 +1,575 @@
...
+  <macro name="author-biblio">
+    <choose>
+      <if type="legal_case" match="any">
+        <text variable="authority"/>
+      </if>
+      <else-if match="any" variable="author">
+        <names variable="author">
+          <name name-as-sort-order="all" sort-separator="&#160;">
+            <name-part name="family" text-case="capitalize-first" font-variant="small-caps"/>
+            <name-part name="given" text-case="capitalize-first" font-variant="normal" prefix="(" suffix=")"/>
```
Comment (rmzelle) [inline sorbonne-student-law-review.csl:94]: @p-heckler, you're encapsulating given names with parentheses here, which is probably not what you want.

File: sorbonne-student-law-review.csl [hunk 2]
```diff
@@ -0,0 +1,575 @@
...
+        </names>
+      </else-if>
+      <else-if match="any" variable="editor collection-editor">
+        <choose>
+          <!-- for chapter or paper conference, no author implies the editor is also the author. Name should remain in editor position-->
+          <if type="chapter paper-conference" match="none">
+            <names variable="editor">
+              <name name-as-sort-order="all" sort-separator="&#160;">
+                <name-part name="family" text-case="capitalize-first" font-variant="small-caps"/>
+                <name-part name="given" text-case="capitalize-first" font-variant="normal" prefix="(" suffix=")"/>
```
Comment (rmzelle) [inline sorbonne-student-law-review.csl:105]: Same here.

File: sorbonne-student-law-review.csl [hunk 3]
```diff
@@ -0,0 +1,575 @@
...
+  </macro>
+  <macro name="editor-biblio">
+    <choose>
+      <if match="any" type="chapter paper-conference entry-encyclopedia entry-dictionary">
+        <group delimiter="&#160;">
+          <text term="in" font-style="italic"/>
+          <names variable="editor collection-editor">
+            <name name-as-sort-order="all" sort-separator="&#160;">
+              <name-part name="family" text-case="capitalize-first" font-variant="small-caps"/>
+              <name-part name="given" text-case="capitalize-first" font-variant="normal" prefix="(" suffix=")"/>
```
Comment (rmzelle) [inline sorbonne-student-law-review.csl:129]: And here.

File: sorbonne-student-law-review.csl [hunk 4]
```diff
@@ -0,0 +1,575 @@
...
+      </if>
+    </choose>
+  </macro>
+  <macro name="translator-biblio">
+    <group delimiter="&#160;">
+      <text term="translator" form="short"/>
+      <names variable="translator">
+        <name name-as-sort-order="all" sort-separator="&#160;">
+          <name-part name="family" text-case="capitalize-first" font-variant="small-caps"/>
+          <name-part name="given" text-case="capitalize-first" font-variant="normal" prefix="(" suffix=")"/>
```
Comment (rmzelle) [inline sorbonne-student-law-review.csl:143]: And here.

**Comment by rmzelle** [comment]
> Somehow, the generated citations here do not match the output I obtain when I test the file. ... Again, I don't have this problem when testing it locally and I'm not sure where the code goes wrong here.

The citation previews here suffer from a few formatting bugs, and you shouldn't base your style on them. How are you testing your style locally? Zotero?

You probably want to fix the issue I highlighted, though, as I didn't see any example citations in https://sorbonnestudentlawreview.org/charters/charter_EN where given names are put in parentheses.

**Comment by p-heckler** [comment]
Hi @rmzelle! 
Thank you very much for your input! 
I have fixed the given name encapsulation issue. Since the Sorbonne Law Review does not specify bibliographical entries, I was using the Sorbonne publishing guidelines, which the style mentions as a fallback, and the version I was using had given names encapsulated. That version is no longer online as of this week (they are migrating their website) and the only version I can find does not contain encapsulation, so I will abide to this last version.

> How are you testing your style locally? Zotero?

Yes, I occasionally use Juris-M, but Zotero is my primary testing method and a few people who are using it from my Github repo and giving me feedback are using Zotero as well.

**Comment by rmzelle** [comment]
Thanks!

**Pull request merged**